### ISSUE TITLE
Asset change is sent to the unknown address

### ISSUE THREAD

**Issue by UkolovaOlga** [RGB-WG/rgb-wallet#53]
#### Description
We followed the RGB v0.10 release video demo, generated all our wallets using descriptor wallet, managed to issue contracts, perform a transfer and update our stash. However when we broadcast the final PSBT to the blockchain the change UTXO was sent to an address that the sending wallet did not control and this effectively resulted in us transfer 100 of an RGB20 asset but having all the change (so the remaining supply) sent to a random address. We can recreate the issue every time we try transfer. No errors are produced.

The bug is also present when calling “rgb utxos” after broadcasting the transfer as the RGB CLI can’t detect the UTXO either.

In the video Maxim uses this TPUB: tpubDCNiWHaiSkgnQjuhsg9kjwaUzaxQjUcmhagvYzqQ3TYJTgFGJstVaqnu4yhtFktBhCVFmBNLQ5sN53qKzZbMksm3XEyGJsEhQPfVZdWmTE2

Then the following steps are made:
- create an RGB wallet for with the descriptor tapret(tpubDCNiWHaiSkgnQjuhsg9kjwaUzaxQjUcmhagvYzqQ3TYJTgFGJstVaqnu4yhtFktBhCVFmBNLQ5sN53qKzZbMksm3XEyGJsEhQPfVZdWmTE2, ...)
- issue 1000000000000000 of an RGB20 asset called TST to a UTXO associated with this wallet.
- perform a transfer action where he transfers 100 TST to a new UTXO, sign the transfer PSBT and then broadcast it.

This is the transaction he broadcast in the video: https://www.blockstream.info/testnet/tx/85f9b8ce3b495a206de334928c89ce8fc93f02304b634641ac76e4008948e2a3

The transaction has 1 input (The UTXO which sealed the TST) and 1 output, which is the change from the transaction. The change also is the UTXO where the remaining TST is bound. The address the change is sent to is: tb1pmv024fd3f62peg5rg5ex8m3zl4n9qkze5xsrw58fknh5987hkmusflwcpu

The tpub from earlier is unable to detect the change utxo (85f9b8ce3b495a206de334928c89ce8fc93f02304b634641ac76e4008948e2a3:0) when calling "rgb utxos" OR when calling "btc-cold check <WALLETFILE>" furthermore the change utxo is unspent (meaning the problem is not that it has been spent). 

The result in the video and when we performed tests is that when you do a RGB transfer call the correct amount of assets get transferred to the person you want to send them to but the change gets sent to a random address meaning the original owner loses their RGB20 assets (or loses the ability to spend them). From what I can tell this looks like when the RGB CLI is manipulating the PSBT (through the transfer call) it is deriving the change address incorrectly from the tpub.

**Comment by crisdut** [comment]
Hi @UkolovaOlga,

I think this is not necessary a bug, only missing steps in rgb-cli + adjust compatibilities between tools, to facilitate use by people unfamiliar with the projects.

> We followed the RGB v0.10 release video demo, generated all our wallets using descriptor wallet, managed to issue contracts, perform a transfer and update our stash. However when we broadcast the final PSBT to the blockchain the change UTXO was sent to an address that the sending wallet did not control and this effectively resulted in us transfer 100 of an RGB20 asset but having all the change (so the remaining supply) sent to a random address. We can recreate the issue every time we try transfer. No errors are produced.

There is not "random address", just an output script pubkey + tweak. 

In the bitmask project I created a few more steps to allow "spending change contracts", below:

- I created a way to extract the tweak from the PSBT file, so that I can use it to generate a new PSBT.
- I store the tweak (tapret) in RgbDesc, which is a descriptor Maxim created exclusively for RGB CLI
- Customized the descriptor-wallet to support a input + tweak.

I opened an issue on bitmask-core to track all my changes and create "official" fixes or at least help Maxim with testing, hehehe!

I plan to address some in the next few days, I'm reviewing the implementations to remove "customization overkill" =)

See here: https://github.com/diba-io/bitmask-core/issues/204

I hope I've helped.
